### DISCUSSION TITLE
Add a button to create a prompt from a chat message

### DISCUSSION THREAD

**Discussion by jumski** [open-webui/open-webui#3141] [Ideas]
Loving openwebui, I'm selfhosting it for me and friends and want to contribute to learn codebase and scratch my own itch.

Currently when I write a good prompt in new conversation and i want to repeat it for some other input i need to perform 5 actions to get prompt content into "new prompt form":
- copy the message content
- click "Workspace"
- click "Prompts"
- click plus icon to open form
- paste message

I propose to add a button next to edit/copy, below message, that will reduce number of steps to one:

- click "convert to prompt" button below any message

### Use cases

- developing prompts organically by first talking to see the results immediately.
- when you forget you wanted to start with a new prompt to reuse it for next items in the list, and instead already having a good conversation going
- using LLMs to generate prompts (this workflow would be massively improved!)

#### Loose thoughs

- saving prompt should go back to conversation - staying on the prompts page would discourage usage as additional step is necessary to go back to the original conversation

Would be interested in providing PR for this feature if it will be considered useful.

**Comment by tjbck** [comment]
![shrek-shrek-rizz](https://github.com/open-webui/open-webui/assets/25473318/bf71fb6c-1ecd-403a-b7c1-87a7e6dd51f3)

Great suggestion! I'll see what can be done!
